### PR TITLE
[CPM-86] programme regulations ui

### PR DESCRIPTION
**Pull request by anian100**
- UI for adding and displaying programme requirements

### CODE REVIEW

File: client/src/pages/ExistingProgramme/editRequirementsBox.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import React from "react";
```
Comment (vee-christella) [inline client/src/pages/ExistingProgramme/editRequirementsBox.js:1]: Can we rename components files to be capitalised please :)

File: client/src/pages/ExistingProgramme/index.js [hunk 2]
```diff
@@ -30,15 +32,22 @@ const ExistingProgramme = ({ id }) => {
                 </Text>
             </Flex>
 
-            <Flex align="center" justify="center" marginTop="50px">
+            <Flex align="center" justify="center" direction="column">
                 <Text textAlign="center" fontSize="md" color="#000000" as="i">
                     Requirements
                 </Text>
-                <Box as="span" ml="2" color="gray.600" bg="gray" />
+                <Box bg="#e2e2e2" width="75%" textAlign="center">
```
Comment (vee-christella) [inline client/src/pages/ExistingProgramme/index.js:39]: Can we please add some padding/margin so it's not so close to the title?

File: client/src/pages/ExistingProgramme/index.js [hunk 3]
```diff
@@ -30,15 +32,22 @@ const ExistingProgramme = ({ id }) => {
...
+                    <IconButton width="90%" icon="add" bg="#616161" color="white" onClick={() => setCreateNewRequirements(true)} />
```
Comment (vee-christella) [inline client/src/pages/ExistingProgramme/index.js:40]: Same as above, let's add some more padding so the button doesn't touch the top of the box

File: client/src/pages/ExistingProgramme/editRequirementsBox.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            alignSelf="center"
+            direction="column"
+            textAlign="flex-start"
+            marginTop="20px"
+            height="150px"
+            width="85%"
+            color="white"
+            bg="#e2e2e2"
+        >
+            <Text textAlign="center" fontSize="md" color="#000000" as="i">
```
Comment (vee-christella) [inline client/src/pages/ExistingProgramme/editRequirementsBox.js:29]: Extra padding at the top again would be great :D

**Review by vee-christella — APPROVED** [review]
Just some minor padding comments that we should do now since it's just UI related. Also scrollbar looks good 🌵 

Can you also create a bug ticket for the weird dropdown arrow?

File: client/src/pages/ExistingProgramme/editRequirementsBox.js (hunk 4 above)
Comment (anian100) [reply client/src/pages/ExistingProgramme/editRequirementsBox.js]: done

File: client/src/pages/ExistingProgramme/index.js (hunk 3 above)
Comment (anian100) [reply client/src/pages/ExistingProgramme/index.js:40]: done

File: client/src/pages/ExistingProgramme/index.js (hunk 2 above)
Comment (anian100) [reply client/src/pages/ExistingProgramme/index.js]: done

File: client/src/pages/ExistingProgramme/editRequirementsBox.js (hunk 1 above)
Comment (anian100) [reply client/src/pages/ExistingProgramme/editRequirementsBox.js]: done

**Review by wryao64 — APPROVED** [review]
Left some comments that might help with refactoring but not super important. Otherwise LGTM

**Comment by FocalChord** [comment]
Good work on the code!

Just a few minor (non blocking) design things, this down arrow is a bit weird

Screen Shot 2020-09-12 at 12 24 24 PM

The outline on these pills are too tightly bound to the text, also could we change the outline color to something else more visible? Maybe white


Screen Shot 2020-09-12 at 12 24 54 PM

What are your thoughts on this?

**Comment by anian100** [comment]
Can't reply to @FocalChord 's comment above directly so I'll write my thoughts here.
I have been trying to fix that arrow for a while now but not sure how to do it. So I'll make a ticket for now to be fixed later. Agree that the outline of the text is a bit tight. I can try to fix that now. I also agree that the outline is not quite visible, I'll change it to white for now. Designers feel free to change it back if you disagree

**Comment by daltonlim** [comment]
> Screen Shot 2020-09-12 at 12 24 54 PM
> 
> What are your thoughts on this?

Doesn't cpm-84 have a similar course pill. Could probably standardize that component and reuse it.

**Pull request merged**